controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1611,6 +1616,88 @@ bool master_thread::handle_cmdu_1905_topology_notification(const std::string &sr
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_topology_response(const std::string &src_mac,
+                                                       ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received TOPOLOGY_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto tlvDeviceInformation = cmdu_rx.getClass<ieee1905_1::tlvDeviceInformation>();
+    if (!tlvDeviceInformation) {
+        LOG(ERROR) << "ieee1905_1::tlvDeviceInformation not found";
+        return false;
+    }
+
+    const auto &al_mac = tlvDeviceInformation->mac();
+
+    // Get all Agent known fronthaul radios if exist, and compare it to what the Agent reports.
+    // If a radio exist on the database but not the tlvDeviceInformation, this radio node needs
+    // to be removed.
+    // This shall work in the opposite way as well: If it doesn't exist on the database, but
+    // reported in the TLV, then we need to add the node to the database. Of course this will remain
+    // as "TODO" for future task. Meanwhile parse properly the TLV, leaving the unused parts in
+    // comment for future implementation.
+    auto fronthaul_radios_on_db = database.get_node_children(
+        network_utils::mac_to_string(al_mac), beerocks::TYPE_SLAVE, beerocks::STATE_CONNECTED);
+
+    std::unordered_set<std::string> reported_fronthaul_radios;
+
+    for (uint8_t i = 0; i < tlvDeviceInformation->local_interface_list_length(); i++) {
+        const auto iface_info_tuple = tlvDeviceInformation->local_interface_list(i);
+        if (!std::get<0>(iface_info_tuple)) {
+            LOG(ERROR) << "Failed to get " << int(i) << " element of local iface info "
+                       << "on Device Information TLV";
+            return false;
+        }
+
+        auto &iface_info = std::get<1>(iface_info_tuple);
+
+        const auto &iface_mac = iface_info.mac();
+        auto iface_mac_str    = network_utils::mac_to_string(iface_mac);
+
+        // For future implementation
+        // const auto media_type        = iface_info.media_type();
+
+        // For wireless interface it is defined on IEEE 1905.1 that the size of the media info
+        // is n=10 octets, which the size of s802_11SpecificInformation struct.
+        // For wired interface n=0.
+        if (iface_info.media_info_length() == 10) {
+            const auto media_info = reinterpret_cast<ieee1905_1::s802_11SpecificInformation *>(
+                iface_info.media_info(0));
+            const auto iface_role = media_info->role;
+
+            // For future implementation
+            // const auto &iface_bssid = media_info->network_membership;
+            // const auto iface_bw     = media_info->ap_channel_bandwidth;
+            // const auto iface_cf1    = media_info->ap_channel_center_frequency_index1;
+            // const auto iface_cf2    = media_info->ap_channel_center_frequency_index2;
```

Comment:
```suggestion
            // In MultiAP context, bssid, bw, cf1 and cf2 are not actually useful since they are more accurately
            // represented in other TLVs (operating channel report, AP operational BSS). 
```

Reply:
AFAIR, the current channel selection task does use it. I'm not sure that we will not use it. Keep it for now.

Reply:
Our mindset is to delete what we don't use **now** - so I don't see why we should keep it. In addition, the channel selection task will be stripped down to nothing very soon so we don't really care if it uses it since its obsolete.

Reply:
Channel selection task will use this information - but coming from the AP Radio Basic Capabilities and the Channel Preference Report, which is a lot more accurate than the limited information given in this TLV.

Reply:
After further discusion - it turns out that p_channel_center_frequency_index2 is missing from the Operating Channel Report TLV, so we *will* need to use this for 80+80MHz operation... Which is super-weird...